src/entries/popup/pages/hw/gridplus/addressChoice.tsx

```diff
@@ -0,0 +1,160 @@
+import { getAddress } from '@ethersproject/address';
+import { motion } from 'framer-motion';
+import { fetchAddresses } from 'gridplus-sdk';
+import { FormEvent, useEffect, useState } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { Address } from 'wagmi';
+
+import { i18n } from '~/core/languages';
+import { SessionStorage } from '~/core/storage';
+import { truncateAddress } from '~/core/utils/address';
+import { Box, Button, Text } from '~/design-system';
+import { Checkbox } from '~/entries/popup/components/Checkbox/Checkbox';
+import { Link } from '~/entries/popup/components/Link/Link';
+import { Spinner } from '~/entries/popup/components/Spinner/Spinner';
+import { HARDWARE_WALLETS } from '~/entries/popup/handlers/walletVariables';
+import { useAccounts } from '~/entries/popup/hooks/useAccounts';
+import { ROUTES } from '~/entries/popup/urls';
+
+export type AddressesData = {
+  addresses: Address[];
+};
+
+export const AddressChoice = () => {
+  const navigate = useNavigate();
+  const { state } = useLocation();
+  const { sortedAccounts } = useAccounts();
+  const [formData, setFormData] = useState({
+    selectedAddresses: [] as string[],
+  });
+  const [loadingAddresses, setLoadingAddresses] = useState(true);
+  const [addresses, setAddresses] = useState<AddressesData['addresses']>([]);
+  const disabled = loadingAddresses || formData.selectedAddresses.length === 0;
+  const toggleAddress = (address: string) => {
+    const selected = formData.selectedAddresses.includes(address);
+    if (selected)
+      return setFormData({
+        selectedAddresses: formData.selectedAddresses.filter(
+          (currentAddress) => currentAddress !== address,
+        ),
+      });
+    return setFormData({
+      selectedAddresses: [...formData.selectedAddresses, address],
+    });
+  };
+  const onSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const accountsToImport = formData.selectedAddresses.map((address, i) => ({
+      address,
+      index: i,
```

Comment:
is there a way to "add by index"? this index here should be the index of the wallet in the derivation path, not it's index in the array

Reply:
The index in this view actually relates to the derivation path in Lattice. If there's no override for `fetchAddresses` starting path, then it starts from address index 0.